.github/workflows/test.yml

```diff
@@ -0,0 +1,31 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+            dex/target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: test
+        run: |
+          cargo test --manifest-path Cargo.toml --all
+          cargo test --manifest-path dex/Cargo.toml --all
```

Comment:
Why do we need this second line?

Reply:
the root cargo toml excludes the program:

```
[workspace]
members = [
    "assert-owner",
    "common",
    "dex/crank",
    "dex/permissioned",
]
exclude = [
    "dex",
]
```